src/frequenz/repo/config/mkdocs/sort_versions.py

```diff
@@ -0,0 +1,159 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Sort `mike`'s `version.json` file with a custom order.
+
+The `version` keys are expected as follows:
+
+- `vX` and `vX.Y` for tagged versions
+- `vX-dev` and `vX.Y-dev` for development versions
+- Any other arbitrary string for other versions
+
+The sorting order is as follows:
+
+- Other versions appear first and are sorted alphabetically between them.
+- Development versions appear before tagged versions for the same `vX` and `vX.Y`
+   version.
+- Regular semver sorting is used within the same tagger or development version.
+"""
+
+import functools
+import json
+import re
+import sys
+from typing import TypedDict
+
+from semver.version import Version as SemVersion
+
+_is_version_re = re.compile(r"^v(\d+)")
+_major_to_semver_re = re.compile(r"^v(\d+)-dev$")
+_minor_to_semver_re = re.compile(r"^v(\d+).(\d+)-dev$")
+_patch_to_semver_re = re.compile(r"^v(\d+).(\d+)$")
+
+
+class Version(TypedDict):
+    """A version entry in `version.json`."""
+
+    version: str
+    """The version string."""
+
+    title: str
+    """The version title."""
+
+    aliases: list[str]
+    """The version aliases."""
+
+
+def to_semver(version: str) -> str:
+    """Convert a version string to a semver string.
+
+    The following transformations are applied:
+
+    - `vX` -> `vX.9999.9999-zzzz`
+    - `vX.Y` -> `vX.Y.9999-zzzz`
+    - `vX.Y.Z` -> `vX.Y.Z`
```

Comment:
I assume the `999`s and the `zz`s are there to make it sort them to the end(or front depending on order)?
Maybe a comment mentioning that part would be helpful, strictly speaking this is more "orderable_semver" no?

Reply:
Yeah, so the new function name should make it more evident, but I'll improve the description a bit more too.